app/controllers/memes_controller.rb

```diff
@@ -65,6 +65,76 @@ def destroy
     end
   end
 
+  def upvote
+    @vote = Vote.where(meme_id: params[:id], user: current_user)
+
+    if @vote.count != 0
+      @vote_value = @vote[0].value
+      @vote.destroy_all
+
+      @created_upvote = false;
+      @deleted_downvote = false;
+
+      if @vote_value == false
+        Vote.create(meme_id: params[:id], user: current_user, value: true)
+        @created_upvote = true;
+        @deleted_downvote = true;
+      end
+    else
+      Vote.create(meme_id: params[:id], user: current_user, value: true)
+      @created_upvote = true;
+      @deleted_downvote = false;
+    end
+
+    # Return value
+    #   created_upvote: true if upvote was created, false if deleted
+    #   deleted_downvotedownvote: true if downvote was deleted, false if there was no downvote
+    #
+    respond_to do |format|
```

Comment:
no need for a respond_to block when you're always returning json; just do the `render :json`